x-pack/plugins/observability_solution/logs_explorer/public/state_machines/logs_explorer_controller/src/services/selection_service.ts

```diff
@@ -35,26 +39,44 @@ export const initializeSelection =
   async (send) => {
     /**
      * First validation.
-     * If the selection is a data view which is not of logs type, redirect to Discover.
-     */
-    if (
-      isDataViewSelection(context.datasetSelection) &&
-      context.datasetSelection.selection.dataView.isUnknownDataType()
-    ) {
-      return redirectToDiscover({ context, datasetSelection: context.datasetSelection, discover });
-    }
-
-    /**
-     * Second validation.
-     * If the selection is a data view, initialize it.
+     * The selection is a data view.
      */
     if (isDataViewSelection(context.datasetSelection)) {
-      return send('INITIALIZE_DATA_VIEW');
+      let datasetSelection: DataViewSelection | null = context.datasetSelection;
```

Comment:
**note**: this refactor updates the initialization flow such that:
1. If the data view selection is unresolved, perform a data view look-up to retrieve it by id:
  - if the data view does not exist, fallback to the default selection.
  - if the data view exists, continue the assertions
2. If the data view is a non-logs one, redirect to Discover.
3. Final case is the data view is about logs, so load it in LogsExplorer